tests/torch/fx/test_models.py

```diff
@@ -126,33 +147,51 @@ def test_model(test_case: ModelCase):
     ("model_case", "quantization_parameters"), TEST_MODELS_QUANIZED, ids=[m[0].model_id for m in TEST_MODELS_QUANIZED]
 )
 def test_quantized_model(model_case: ModelCase, quantization_parameters):
-    with disable_patching():
-        model = model_case.model_builder()
-        example_input = torch.ones(model_case.input_shape)
-
-        with torch.no_grad():
-            model.eval()
-            fx_model = capture_pre_autograd_graph(model, args=(example_input,))
-
-        def transform_fn(data_item):
-            return data_item.to("cpu")
-
-        calibration_dataset = nncf.Dataset([example_input], transform_fn)
-
-        quantization_parameters["advanced_parameters"] = AdvancedQuantizationParameters(disable_bias_correction=True)
-        quantization_parameters["subset_size"] = 1
-
-        quantized_model = nncf.quantize(fx_model, calibration_dataset, **quantization_parameters)
-        # Uncomment to visualize torch fx graph
-        # from tests.torch.fx.helpers import visualize_fx_model
-        # visualize_fx_model(quantized_model, f"{model_case.model_id}_int8.svg")
-
-        nncf_graph = GraphConverter.create_nncf_graph(quantized_model)
-        check_graph(nncf_graph, get_dot_filename(model_case.model_id), FX_QUANTIZED_DIR_NAME)
+    model = model_case.model_builder()
+    dtype = torch.int32 if model_case.model_id == "synthetic_transformer" else torch.float32
+    example_input = torch.ones(model_case.input_shape, dtype=dtype)
+
+    with torch.no_grad():
+        model.eval()
+        fx_model = _capture_model(model, example_input)
+
+    def transform_fn(data_item):
+        return data_item.to("cpu")
+
+    calibration_dataset = nncf.Dataset([example_input], transform_fn)
+
+    quantization_parameters["advanced_parameters"] = AdvancedQuantizationParameters(disable_bias_correction=True)
+    quantization_parameters["subset_size"] = 1
+
+    quantized_model = nncf.quantize(fx_model, calibration_dataset, **quantization_parameters)
+    # Uncomment to visualize torch fx graph
+    # from tests.torch.fx.helpers import visualize_fx_model
+    # visualize_fx_model(quantized_model, f"{model_case.model_id}_int8.svg")
+
+    nncf_graph = GraphConverter.create_nncf_graph(quantized_model)
+    check_fx_graphs(nncf_graph, get_dot_filename(model_case.model_id), FX_QUANTIZED_DIR_NAME)
+
+
+@pytest.mark.parametrize("unification", [False, True])
+def test_is_shared_attribute(unification):
+    model = MultiBranchesConnectedModel()
+    ex_inputs = torch.ones((1, 3, 3, 3))
+    captured_model = _capture_model(model, ex_inputs)
+    file_prefix = "not_unified"
+    if unification:
+        file_prefix = "unified"
+        shared_constants_unification_transformation(captured_model)
+    nncf_graph = GraphConverter.create_nncf_graph(captured_model)
+    shared_attributes = {n.node_name: n.is_shared() for n in nncf_graph.get_all_nodes()}
+    get_ref_from_json(f"{file_prefix}_shared_attribute_test_model", shared_attributes, attributes=True)
```

Comment:
Assert?

Reply:
Sorry about that it got mixed up in the commits. I am pushing the updated code now!